### PR TITLE
feat(config): support a configuration file

### PR DESCRIPTION
**Pull request by screendriver**
It is now possible to create a `glpdrc.js` configuration file **in addition** to command line arguments.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,3 +23,15 @@ Options:
   --trace           show stack traces for errors when possible (default: false)
   -h, --help        display help for command
 ```
+
+You can use either the command line arguments `gitlab-url`, `project-id` and `access-token` or you can create a `glpdrc.js` configuration file that exports an object. All of the command line arguments are supported but needs to be written in `camelCase`:
```
Comment (lo1tuma) [inline README.md:27]: Personally I would prefer `glpd.config.js`.

Comment (screendriver) [reply README.md]: 👌

File: src/bin.tsx [hunk 2]
```diff
@@ -16,36 +19,61 @@ program
...
-        exit={exit}
-        listPipelines={listPipelinesFunction}
-        filterPipelinesByDate={filterPipelinesByDate}
-        deletePipeline={deletePipelineFunction}
-        showStackTraces={options.trace === true}
-      />,
-    );
-  });
+  .action(
+    async (
```
Comment (lo1tuma) [inline src/bin.tsx:28]: I think for `async` actions it is required to use `program.parseAsync` instead of `program.parse`.

File: src/bin.tsx [hunk 3]
```diff
@@ -16,36 +19,61 @@ program
...
+    ) => {
+      const cliArguments: CliArguments = {
+        gitlabUrl: gitlabUrlArgument,
+        projectId: projectIdArgument ? parseInt(projectIdArgument, 10) : undefined,
+        accessToken: accessTokenArgument,
+        days: typeof options.days === 'string' ? parseInt(options.days, 10) : 30,
+        trace: options.trace === true,
+      };
+      const explorer = cosmiconfig('gitlab-pipeline-deleter');
+      const config = await loadConfig('./glpd.config.js', explorer);
```
Comment (lo1tuma) [inline src/bin.tsx:42]: Does it automatically load the file relative to the CWD?

File: src/bin.tsx [hunk 4]
```diff
@@ -16,36 +19,61 @@ program
...
+      const glpdArguments = mergeCliArgumentsWithConfig(cliArguments, config);
```
Comment (lo1tuma) [inline src/bin.tsx:43]: So when I define e.g. `gitlabUrl` via CLI *and* config file, then the config file has precedence, right?

* I think this this should be documented
* Maybe it is better to prefer the CLI arguments in case you want to overwrite the config in the config file (e.g. useful in CI pipelines, where need to run the tool for an edge-case scenario with a different setting). WDYT?

Comment (lo1tuma) [reply src/bin.tsx:43]: BTW: The precedence order should be also covered by unit tests.

File: src/bin.tsx (hunk 2 above)
Comment (screendriver) [reply src/bin.tsx:28]: Good point 👍 I didn't knew about that function.

File: src/bin.tsx (hunk 4 above)
Comment (lo1tuma) [reply src/bin.tsx:43]: Never mind, seems like most of this is already implemented and tested. I was confused by the argument order here because other merge functions merge the arguments from right to left.

But I think it would be still good to document this behavior.

Comment (screendriver) [reply src/bin.tsx:43]: It's the other way around. The CLI arguments are preferred over configuration file arguments. See also the unit test `mergeCliArgumentsWithConfig() prefers CLI arguments over configuration values`

I will update the documentation about this behaviour.

File: src/bin.tsx (hunk 3 above)
Comment (screendriver) [reply src/bin.tsx:42]: Good question 🤔 Would it make more sense to have a `path.resolve(__dirname, 'glpd.config.js')` here?

File: src/bin.tsx (hunk 4 above)
Comment (screendriver) [reply src/bin.tsx:43]: > BTW: The precedence order should be also covered by unit tests.

It is in `mergeCliArgumentsWithConfig() prefers CLI arguments over configuration values`

Comment (screendriver) [reply src/bin.tsx:43]: > other merge functions merge the arguments from right to left

True. I did this because of potential currying or composition where the last argument is always that one that is used for the function application.

File: src/bin.tsx (hunk 3 above)
Comment (lo1tuma) [reply src/bin.tsx:42]: No, `__dirname` doesn’t make sense, since you want to load the file relative to the user CWD folder and not the `gitlab-pipeline-delete` folder within `node_modules`.

Comment (screendriver) [reply src/bin.tsx:42]: Ah sorry I meant `process.cwd()`. So `path.resolve(process.cwd(), 'glpd.config.js')` ?

**Review by lo1tuma — APPROVED** [review]
GNTM 👗
